internal/build/lifecycle_execution.go

```diff
@@ -720,6 +740,67 @@ func (l *LifecycleExecution) ExtendRun(ctx context.Context, kanikoCache Cache, p
 	return extend.Run(ctx)
 }
 
+const (
+	argBuildID = "build_id"
+	argUserID  = "user_id"
+	argGroupID = "group_id"
+)
+
+/*
+	This implementation of ExtendBuildByDaemon is based on the RFC #0105 which uses docker daemon to extend the build Image instead of kaniko.
+	* Parsing the `group.toml` from the temp directory of buildpack and set the extensions.
+	* Reading the dockerfiles that were generated during the `generate` phase and also parsing the Arguments given by the user from
+	`extend-config.toml`.
+	* Using ImageBuild method of docker API client to extend the Image and save it to the daemon.
+	* Invoking Build phase of lifecycle by creating a container from the extended Image and dropping the privileges.
+
+*/
+
+func (l *LifecycleExecution) ExtendBuildByDaemon(ctx context.Context) error {
+	builderImageName := l.opts.BuilderImage
+	extendedBuilderImageName := l.opts.BuilderImage + "-extended"
+	defaultFilterFunc := func(file string) bool { return true }
```

Comment:
nit: can we make this an anonymous function, or move it closer to where it's being used, so that it's easier for the reader?

Reply:
done